### PR TITLE
Add select all option for export

### PR DESCRIPTION
**Pull request by jaads**
Added the option to select all fields at once when exporting a collection.
The option can be selected on the root fields and on the nested fields.

Here is a small demo:

https://user-images.githubusercontent.com/34009329/231815937-7ee33548-e87e-49d9-b4b7-9787a9949280.mov

I tried to add a divider below the two new options but I thought it was not very good looking for the nested groups so I left it very simple. But any ideas are welcome!

For the select all option I used the existing emitter to populate the change to the parent component instead of creating a new one. Now the event should always contain an array - even for a single item. This way the code gets less bloated and we don't have much redundancy. 

I tried to make write some tests for this feature, but to test this in isolation was very difficult, since the components has a lot of references to other components which were hard to mock.

Closes ENG-454

### CODE REVIEW

**Comment by Nitwel** [comment]
This is looking awesome!
Some small improvements we could do:
1. Only ever show 1 button, when not all are selected show all, when all are selected show deselect all.
2. It would be nice to have a divider between the options in order to more easily distinguish between them.

**Review by br41nslug — COMMENTED** [review]
Looking good but should "deselect all" in the root not also deselect relational fields?

https://user-images.githubusercontent.com/9389634/232077862-e6e205b3-6605-4c88-ad37-f1aad17c8ef4.mp4

**Comment by rijkvanzanten** [comment]
@br41nslug I'll do you one better; why do we even have a deselect-all option? I don't think that was in scope for ENG-454 😄

**Comment by br41nslug** [comment]
Well it does make sense to me that when adding a "select all" option because manually selecting all options is too much clicking that the reverse would be present or wanted too.

**Comment by rijkvanzanten** [comment]
> Well it does make sense to me that when adding a "select all" option because manually selecting all options is too much clicking that the reverse would be present or wanted too.

The difference is that selecting a field is a multi-click (click "Add Field", click the relational field recursive, click the field) for each field to add, whereas the deselect is a `x` that's always present. It's way easier to quickly deselect multiple fields than it is to quickly select multiple fields 👍🏻

**Comment by rijkvanzanten** [comment]
Also in the UX content of this dropdown, "Select All" applies to the options present in the dropdown, whereas "Deselect All" relates to the curren top-level selection (outside of the dropdown context). If we wanted to have a "Deselect All", it should be on the parent list-level rather than in the dropdown 👍🏻 

I'd suggest we drop deselect-all for now from this PR, and revisit it in a new PR on the top-level, if it's something we want to have 🙂

File: app/src/components/v-field-list/v-field-list-item.vue [hunk 1]
```diff
@@ -33,18 +33,28 @@
...
+		</template>
+
 		<v-field-list-item
 			v-for="childField in field.children"
 			:key="childField.key"
 			:field="childField"
 			:search="search"
 			:include-functions="includeFunctions"
 			:relational-field-selectable="relationalFieldSelectable"
-			@add="$emit('add', $event)"
```
Comment (rijkvanzanten) [inline app/src/components/v-field-list/v-field-list-item.vue:43]: This erroneously broke selecting individual fields

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: app/src/interfaces/_system/system-fields/system-fields.vue [hunk 2]
```diff
@@ -28,7 +28,13 @@
 				</button>
 			</template>
 
-			<v-field-list :disabled-fields="value" :collection="collectionName" @select-field="addField" />
```
Comment (rijkvanzanten) [inline app/src/interfaces/_system/system-fields/system-fields.vue:31]: This event rename is a breaking change that wasn't applied to all other uses of `v-field-select`

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: app/src/interfaces/_system/system-fields/system-fields.vue (hunk 2 above)
Comment (rijkvanzanten) [reply app/src/interfaces/_system/system-fields/system-fields.vue:31]: Also changing the event type from `string` to `string[]` makes it a bit of a tricky refactor 😁

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**